consensus/src/dag/dag_store.rs

```diff
@@ -139,8 +173,39 @@ impl Dag {
         }
     }
 
-    pub fn bitmask(&self) -> Vec<Vec<bool>> {
-        // TODO: extract local bitvec
-        todo!();
+    pub fn lowest_incomplete_round(&self) -> Option<Round> {
+        for (round, round_nodes) in &self.nodes_by_round {
+            if round_nodes.iter().any(|node| node.is_none()) {
+                return Some(*round);
+            }
+        }
+        None
+    }
+
+    pub fn bitmask(&self) -> Option<(Round, Vec<Vec<bool>>, usize)> {
+        let lowest_round = match self.lowest_incomplete_round() {
+            Some(lowest_round) => lowest_round,
+            None => return None,
+        };
+
+        let mut missing_count = 0;
+        let bitmask = self
+            .nodes_by_round
```

Comment:
I think you are not requesting enough.
What if you need to fetch patents in round r, but you have not received nodes in round r-1 yet? (you will not have (r-1, Vec) in the map).